Sources/KituraWebSocket/WebSocketConnection.swift

```diff
@@ -60,9 +63,18 @@ public class WebSocketConnection {
     
     private var messageState: MessageStates = .unknown
     
-    init(request: ServerRequest) {
+    init(request: ServerRequest, service: WebSocketService? = nil) {
         self.request = WSServerRequest(request: request)
         buffer = NSMutableData(capacity: WebSocketConnection.bufferSize) ?? NSMutableData()
+        self.service = service
+        if let connectionTimeout = service?.connectionTimeout {
+            lastFrameReceivedAt = Date()
+            timer = DispatchSource.makeTimerSource()
```

Comment:
timer is a private let on WebSocketConnection and so must be set in the initializer. We could have it as a var that is defaulted to nil but since it is not changed I thought this was better.

Reply:
timerStart requires self and so needs all stored properties to be initialized before being called. This means we can't have it return the timer. We could have the timer as an optional var defaulted to nil but since it is not changed I thought this was better.